tensorboard/embedding.py

```diff
@@ -106,3 +106,70 @@ def add_embedding(mat, save_path, metadata=None, label_img=None):
     make_mat(mat.tolist(), save_path)
     make_pbtxt(save_path, metadata, label_img)
 
+def append_pbtxt(f, metadata, label_img,path):
+
+    f.write('embeddings {\n')
+    f.write('tensor_name: "{}"\n'.format(os.path.join(path,"embedding")))
+    f.write('tensor_path: "{}"\n'.format(os.path.join(path,"tensors.tsv")))
+    if metadata is not None:
+        f.write('metadata_path: "{}"\n'.format(os.path.join(path,"metadata.tsv")))
+    if label_img is not None:
+        f.write('sprite {\n')
+        f.write('image_path: "{}"\n'.format(os.path.join(path,"sprite.png")))
+        f.write('single_image_dim: {}\n'.format(label_img.size(3)))
+        f.write('single_image_dim: {}\n'.format(label_img.size(2)))
+        f.write('}\n')
+    f.write('}\n')
+
+
+class EmbeddingWriter(object):
+    """
+    Class to allow writing embeddings ad defined timestep
+
+    """
+    def __init__(self,save_path):
+        """
+
+        :param save_path: should be the same path of you SummaryWriter
+        """
+        self.save_path = save_path
+        #make dir if needed, it should not
+        try:
+            os.makedirs(save_path)
+        except OSError:
+            print('warning: dir exists')
```

Comment:
Never mind, this will solved after integration. https://github.com/lanpa/tensorboard-pytorch/issues/10#issuecomment-322259372